include/gridtools/fn/sid_neighbor_table.hpp

```diff
@@ -67,10 +63,15 @@ namespace gridtools::fn::sid_neighbor_table {
             const auto origin = sid::get_origin(sid);
             const auto strides = sid::get_strides(sid);
 
-            return {origin, strides};
+            return sid_neighbor_table<IndexDimension,
+                NeighborDimension,
+                MaxNumNeighbors,
+                sid::ptr_holder_type<Sid>,
+                sid::strides_type<Sid>>{
+                origin, strides}; // Note: putting the return type into the function signature will crash nvcc 12.0
```

Comment:
The problem is not visible in CI as we only run some integration tests there for time reasons. The full testsuite is only run on cscs systems, but there we don't have CUDA 12 yet.